pkg/splunk/enterprise/util.go

```diff
@@ -349,45 +413,43 @@ func GetAppListFromS3Bucket(client splcommon.ControllerClient, cr splcommon.Meta
 	var s3Response splclient.S3Response
 	var vol enterprisev1.VolumeSpec
 	var err error
-	var s3Client splclient.S3Client
-	var index int
+	var allSuccess bool = true
 
 	for _, appSource := range appFrameworkRef.AppSources {
-		var volName string
-		// get the volume spec from the volume name
-		if appSource.VolName != "" {
-			volName = appSource.VolName
-		} else {
-			volName = appFrameworkRef.Defaults.VolName
-		}
-
-		index, err = checkIfVolumeExists(appFrameworkRef.VolList, volName)
+		vol, err = GetVolume(client, cr, appSource, appFrameworkRef)
 		if err != nil {
-			scopedLog.Error(err, "Invalid volume name provided. Please specify a valid volume name.", "App source", appSource.Name, "Volume name", volName)
+			allSuccess = false
```

Comment:
Do we want to log the failing volumes here too?

Reply:
This is already done in line 396.